### PR TITLE
 Adding istio types in schema files

### PR DESCRIPTION
**Pull request by vcanaa**
Introducing schema file x-google-marketplace properties:
- Compatibility flag
```yaml
x-google-marketplace:
  istio:
    isSupported: True # Indicates whether the app is compatible with Istio.
```
- Deployer property
```yaml
properties:
- name: istioEnabled
  type: ISTIO_ENABLED # Indicates whether the app is being deployed with Istio.
```

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -400,6 +403,41 @@ properties:
...
+- `True`: The app is compatible with Istio.
+- `False`: The app is NOT compatible with Istio.
+
+```yaml
+properties:
+  # Property definitions...
+required:
+  # Required properties...
+x-google-marketplace:
+  istioCompatibility: True # or False
```
Comment (huyhg) [inline docs/schema.md]: I predict that there will be additional parameters here. Can we make this into a struct:

```
x-google-marketplace:
  istio:
    isSupported: True
    # Future...
    version: '>0.5'
```

Comment (vcanaa) [reply docs/schema.md]: That is a very good idea.

File: marketplace/deployer_util/config_helper.py [hunk 2]
```diff
@@ -334,6 +343,20 @@ def memory(self):
     return self._memory
 
 
+class SchemaIstio:
+  """Accesses top level istio."""
+
+  def __init__(self, dictionary):
+    self._is_supported = dictionary.get('isSupported', None)
+
+    if 'isSupported' in dictionary:
```
Comment (huyhg) [inline marketplace/deployer_util/config_helper.py]: rm?

**Pull request merged**